queue/abstract.lua

```diff
@@ -574,6 +586,24 @@ local function identification_init()
     end
 end
 
+--- Create an expiration fiber to cleanup expired sessions.
+local function create_expiration_fiber()
+    local exp_fiber = fiber.new(function()
+        while true do
+            local cur_time = time()
+            for quuid, exp_time in pairs(inactive_sessions) do
+                if cur_time >= exp_time then
+                    cleanup_session_tasks(quuid)
```

Comment:
Switching to read-only will kill the fiber (and moving back to read-write will not run it again). We can add a guardian fiber, which will restart the expiration fiber (like expirationd does). `pcall()` should be sufficient too.

Reply:
Done.